lib/modules/platform/codecommit/util.ts

```diff
@@ -0,0 +1,46 @@
+import is from '@sindresorhus/is';
+import type { Credentials } from 'aws4';
+import * as aws4 from 'aws4';
+import { REPOSITORY_UNINITIATED } from '../../../constants/error-messages';
+
+export function getCodeCommitUrl(
+  region: string,
+  repoName: string,
+  credentials: Credentials
+): string {
+  const signer = new aws4.RequestSigner(
```

Comment:
isn't this also part of the newer AWS sdk?

Reply:
its more like a util for the SDK, but since it uses the SDK creds i can move it to the sdk file i guess